### PR TITLE
Node crone mohamad

### PR DESCRIPTION
**Pull request by mohamadaboria**
"adding cron job function and fixing timezone to winter clock"

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -35,9 +38,22 @@ index.use(
   })
 );
 
-// Routes goes here!
+// // this api to test node crone work
```
Comment (shir149) [inline index.js:41]: delete comments

File: middleware/cron.job.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+const cron = require("node-cron");
```
Comment (shir149) [inline middleware/cron.job.js:1]: you dont work with the updated version of develop,cronjob.js is no longer at middleware folder, its in service

File: middleware/cron.job.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            const start = new Date(experiment.duration.startTime);
+            const end = new Date(experiment.duration.endTime);
+
+
+            console.log("start", start)
+            console.log("end", end)
+            
+            console.log("currenttime",currentTime)
+            console.log(experiment._id)
+            // start.setHours(...start.split(':'))
```
Comment (shir149) [inline middleware/cron.job.js:23]: delete comments

File: index.js [hunk 4]
```diff
@@ -13,12 +14,13 @@ const { errorHandler } = require("./middleware/errorHandler.mw");
...
 const { experimentRouter } = require("./router/experiment.router");
-const { testRouter } = require("./router/external.routes");
+// const { testRouter } = require("./router/external.routes");
 const { userRouter } = require("./router/user.routes");
 const statsRouter = require("./router/stats.router");
 const {goalRouter} = require("./router/goal.router");
-const { experimentStatusUpdate } = require("./Service/cron.job");
+const {experimentStatusUpdate} = require("./middleware/cron.job");
 
+experimentStatusUpdate();
```
Comment (shir149) [inline index.js:23]: also the file index.js is changed to app.js The code should be updated accordingly

**Review by shir149 — DISMISSED** [review]

File: Service/cron.job.js [hunk 5]
```diff
@@ -2,25 +2,34 @@ const cron = require("node-cron");
...
-    query = {status: "active", duration: {endTime:{$gte: now}, startTime: {$lte: now}}};
-    experiments = await ExperimentRepository.findByQuery(query);
-    experiments.forEach((experiment) => {
-        ExperimentRepository.update(experiment._id, {status: "ended"})
-    });
-})
+const experimentStatusUpdate =  ( ) => {
+    const job = cron.schedule("0 8 * * *", async () => {
+      
+        console.log("doing cron job")
```
Comment (AtomicManic) [inline Service/cron.job.js:9]: remove console.log

File: Service/cron.job.js [hunk 6]
```diff
@@ -2,25 +2,34 @@ const cron = require("node-cron");
...
+        const currentTime = new Date();
+        currentTime.setHours(currentTime.getHours() + 2);
```
Comment (AtomicManic) [inline Service/cron.job.js:11]: great job adjusting the cron hours!

File: Service/cron.job.js [hunk 7]
```diff
@@ -2,25 +2,34 @@ const cron = require("node-cron");
...
+        const experiments = await ExperimentRepository.find();
+        for (const experiment of experiments) {
+            const start = new Date(experiment.duration.startTime);
+            const end = new Date(experiment.duration.endTime);
+
+
```
Comment (AtomicManic) [inline Service/cron.job.js:17]: remove all the spaces

File: Service/cron.job.js [hunk 8]
```diff
@@ -2,25 +2,34 @@ const cron = require("node-cron");
...
+            if (experiment.status === "planned" && currentTime >= start && currentTime <= end) {
+              
+                await ExperimentRepository.update(experiment._id, {status: "active"})
+            } else if (experiment.status === "active" && currentTime >= end) {
+
```
Comment (AtomicManic) [inline Service/cron.job.js:22]: remove all spaces

File: index.js [hunk 9]
```diff
@@ -13,12 +14,13 @@ const { errorHandler } = require("./middleware/errorHandler.mw");
 const logPath = path.join(__dirname, "logs", "http.log");
 const port = process.env.PORT || 3000;
 const { experimentRouter } = require("./router/experiment.router");
-const { testRouter } = require("./router/external.routes");
+// const { testRouter } = require("./router/external.routes");
```
Comment (AtomicManic) [inline index.js:17]: remove comment

File: index.js [hunk 10]
```diff
@@ -35,9 +38,11 @@ index.use(
   })
 );
 
-// Routes goes here!
 
-index.use("/test", testRouter);
```
Comment (AtomicManic) [inline index.js:40]: you cannot delete that. Bring this line back, and remove the comments

**Review by AtomicManic — CHANGES_REQUESTED** [review]
Hi. Good job overall. Wrote some comments, please make the corrections (some are crucial!) and fix the conflicts. LMK when you are done

File: router/experiment.router.js [hunk 11]
```diff
@@ -15,9 +15,9 @@ experimentRouter.get("/allowChangeAttribute/:accountId", experimentController.al
 experimentRouter.get('/features', experimentController.getFeaturesList);
 experimentRouter.get('/:experimentId', experimentController.getExperimentById);
 
-
```
Comment (AtomicManic) [inline router/experiment.router.js:18]: remove space

**Review by AtomicManic — APPROVED** [review]

File: index.js [hunk 12]
```diff
@@ -17,8 +18,9 @@ const { testRouter } = require("./router/external.routes");
 const { userRouter } = require("./router/user.routes");
 const statsRouter = require("./router/stats.router");
 const {goalRouter} = require("./router/goal.router");
-const { experimentStatusUpdate } = require("./Service/cron.job");
+const {experimentStatusUpdate} = require("./Service/cron.job");
```
Comment (shir149) [inline index.js:21]: great

File: index.js [hunk 13]
```diff
@@ -27,7 +29,7 @@ index.use(
   })
 );
 index.use(cookieParser());
-
+index.use(bodyParser.urlencoded({extended:false}))
```
Comment (shir149) [inline index.js:32]: its still index , why?

**Pull request merged**